clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/validators/servergroup/DeployDcosServerGroupDescriptionValidator.groovy

```diff
@@ -23,31 +25,43 @@ class DeployDcosServerGroupDescriptionValidator extends AbstractDcosDescriptionV
     super.validate(priorDescriptions, description, errors)
 
     if (!description.region || description.region.empty) {
-      errors.rejectValue "region", "deployDcosServerGroupDescription.region.empty"
+      errors.rejectValue "region", "${descriptionName}.region.empty"
+    } else if (MarathonPathId.validatePart(description.region.replaceAll(DcosSpinnakerAppId.SAFE_REGION_SEPARATOR, MarathonPathId.PART_SEPARATOR))) {
```

Comment:
I don't think this will validate multipart regions correctly - if the region is "foo/bar" this will fail, since the regex only tests parts, not the full path.